### PR TITLE
:truck: Using SRC Layout

### LINKED ISSUE #300: Using src Layout
didn't know what gitmoji to use since it isn't really for style as it is anything else,
https://hynek.me/articles/testing-packaging/
https://blog.ionelmc.ro/2014/05/25/python-packaging/#the-structure
i wouldn't know if it would be too late to use such a layout or not.

### PR DESCRIPTION
**Pull request by VincentRPS**
### Changes

Closes #300
Supersedes #301 

 ### Check off the following

-   [x] I have tested my changes with the current requirements
-   [x] My Code follows the pep8 code style.

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,2 @@
 websockets>=10.0
-aiohttp>=3.7.4post0,<3.8.0
+aiohttp>=3.7.4post0,<4.0.0
```
Comment (Sigmanificient) [inline requirements.txt:2]: Do pincer has been tested on 4.0.0 of aiohttp?

File: packages/speed.txt [hunk 2]
```diff
@@ -1,4 +0,0 @@
-orjson>=3.5.4
```
Comment (Sigmanificient) [inline packages/speed.txt:1]: Why are you removing the packages extra other requirements files? There also setup when working on the repo

File: packages/img.txt [hunk 3]
```diff
@@ -1 +0,0 @@
-Pillow==8.4.0
```
Comment (Sigmanificient) [inline packages/img.txt:1]: Same for that one

File: setup.py [hunk 4]
```diff
@@ -1,7 +1,67 @@
...
+        "Programming Language :: Python :: 3.9",
+        "Programming Language :: Python :: 3.10",
+        "Framework :: AsyncIO",
+        "Framework :: aiohttp",
+        "Topic :: Internet",
+        "Topic :: Software Development :: Libraries",
+        "Topic :: Software Development :: Libraries :: Python Modules",
+        "Topic :: Utilities",
+    ],
+)
```
Comment (Sigmanificient) [inline setup.py:67]: I wouldn't change the `setup.cfg` to a `setup.py` file as it run literally run anything during the install process, and put a load of uneeded logic within the `setup.py` ifself.

>  ​``setup.cfg``  is a cheekily named Python package which supports providing all of a Python distribution's metadata and build configuration via the ​``setup.cfg``​ ​* file  at the base of the distribution's source tree, rather ​than in the ​``setup.py``​ script. 
    The standard ``setup.py`` script is reduced to a stub which uses the ``setup.cfg`` package to convert the contents of the ``setup.cfg``​ file to keyword arguments understood by the ``setup()``​ function (in particular, the implementation provided by setuptools).  Thus all the standard functionality of ​the ​``setup.py`` script is still supported for tools and procedures that use ​it. 
   ​  In other words, rather than defining your project in a source file, it is defined in an easy to parse, easy to read plain-text file.  However, several hook points are still provided where Python functions can modify the build process if necessary.  But efforts are made to minimize the need for this in the majority of Python projects.

[More](https://docs.python.org/3/distutils/configfile.html)

**Review by Sigmanificient — CHANGES_REQUESTED** [review]
This pr is supposed to change to a src layout but none of the libs files are moved? Moreover, you are changing the setup logic which doesn't seems that much related (as you are moving everything to `setup.py` instead of changing path within `setup.cfg`) and i dont agree with this change

File: requirements.txt (hunk 1 above)
Comment (VincentRPS) [reply requirements.txt:2]: accidently bumped but no and it doesn't install 4.0, will undo because the new pull should do this.

**Comment by Sigmanificient** [comment]
Btw, you reopened the #301 pr, but didn't resolved the conflict as you used an oudated branch?
We had agree on the fact that i will be merged last of every current pr, to avoid creating conflict within them.

File: packages/img.txt (hunk 3 above)
Comment (VincentRPS) [reply packages/img.txt:1]: added to setup.py

**Comment by VincentRPS** [comment]
> Btw, you reopened the #301 pr, but didn't resolved the conflict as you used an oudated branch? We had agree on the fact that i will be merged last of every current pr, to avoid creating conflict within them.

the conflict was with version stuff

**Comment by Sigmanificient** [comment]
> > Btw, you reopened the #301 pr, but didn't resolved the conflict as you used an oudated branch? We had agree on the fact that i will be merged last of every current pr, to avoid creating conflict within them.
> 
> the conflict was with version stuff

The conflict is still  not resolved 

> > Btw, you reopened the #301 pr, but didn't resolved the conflict as you used an oudated branch? We had agree on the fact that i will be merged last of every current pr, to avoid creating conflict within them.
> 
> the conflict was with version stuff

They hasn't been resolved

File: packages/speed.txt (hunk 2 above)
Comment (VincentRPS) [reply packages/speed.txt:1]: just do pip install .[package]

**Comment by VincentRPS** [comment]
> > > Btw, you reopened the #301 pr, but didn't resolved the conflict as you used an oudated branch? We had agree on the fact that i will be merged last of every current pr, to avoid creating conflict within them.
> > 
> > 
> > the conflict was with version stuff
> 
> They hasn't been resolved

fixed.

File: setup.py (hunk 4 above)
Comment (VincentRPS) [reply setup.py:67]: using a setup.cfg and the many differences are just pretty hard imo, this is both easier to read and modify, and most of the pluses of using a setup.cfg instead doesn't seem like anything we would need to care about?

**Pull request closed without merging**